src/corecel/sys/TracingSession.perfetto.cc

```diff
@@ -0,0 +1,136 @@
+//----------------------------------*-C++-*----------------------------------//
+// Copyright 2024 UT-Battelle, LLC, and other Celeritas developers.
+// See the top-level COPYRIGHT file for details.
+// SPDX-License-Identifier: (Apache-2.0 OR MIT)
+//---------------------------------------------------------------------------//
+//! \file corecel/sys/TracingSession.perfetto.cc
+//! \brief RAII class for managing a perfetto session and its resources.
+//---------------------------------------------------------------------------//
+#include "TracingSession.hh"
+
+#include <fcntl.h>
+#include <perfetto.h>
+
+#include "Environment.hh"
+#include "ScopedProfiling.hh"
+
+#include "detail/TrackEvent.perfetto.hh"
+
+PERFETTO_TRACK_EVENT_STATIC_STORAGE();
+
+namespace
+{
+using celeritas::TracingMode;
+
+// Initialize the session for the given mode if profiling is enabled
+std::unique_ptr<perfetto::TracingSession> initialize_session(TracingMode mode)
+{
+    if (!celeritas::use_profiling())
+    {
+        return nullptr;
+    }
+    perfetto::TracingInitArgs args;
+    args.backends |= [&] {
+        switch (mode)
+        {
+            case TracingMode::InProcess:
+                return perfetto::kInProcessBackend;
+            case TracingMode::System:
+                return perfetto::kSystemBackend;
+            default:
+                return perfetto::kSystemBackend;
+        }
+    }();
+    perfetto::Tracing::Initialize(args);
+    perfetto::TrackEvent::Register();
+    return perfetto::Tracing::NewTrace();
+}
+
+// configure the session to record the celeritas track events
+perfetto::TraceConfig configure_session()
+{
+    perfetto::protos::gen::TrackEventConfig track_event_cfg;
+    track_event_cfg.add_disabled_categories("*");
+    track_event_cfg.add_enabled_categories(
+        celeritas::detail::perfetto_track_event_category);
+
+    perfetto::TraceConfig cfg;
+    constexpr int mb_kb = 1024;
+    uint32_t buffer_size_kb = 20 * mb_kb;
+    if (std::string var = celeritas::getenv("CELER_PERFETTO_BUFFER_SIZE_MB");
+        !var.empty())
+    {
+        buffer_size_kb = std::stoul(var) * mb_kb;
+    }
+    cfg.add_buffers()->set_size_kb(buffer_size_kb);
+    auto* ds_cfg = cfg.add_data_sources()->mutable_config();
+    ds_cfg->set_name("track_event");
+    ds_cfg->set_track_event_config_raw(track_event_cfg.SerializeAsString());
+    return cfg;
+}
+//---------------------------------------------------------------------------//
+}  // namespace
+
+namespace celeritas
+{
+
+/*!
```

Comment:
Could you please recheck the merge request for formatting and proper doxygen markup? There are a lot of inconsistencies with the rest of the codebase.